src/main/java/seedu/address/model/person/Patient.java

```diff
@@ -9,28 +9,59 @@
 public class Patient extends Person {
     //fields
     private final Age age;
+    private final Priority priority;
 
     /**
      * Every field must be present and not null.
      */
-    public Patient(Type type, Nric nric, Name name, Phone phone, Age age) {
+    public Patient(Type type, Nric nric, Name name, Phone phone, Age age, Priority priority) {
         super(type, nric, name, phone);
-        requireAllNonNull(age);
+        requireAllNonNull(age, priority);
         this.age = age;
+        this.priority = priority;
     }
 
     public Age getAge() {
         return age;
     }
 
+    public Priority getPriority() {
+        return priority;
+    }
+
     @Override
     public String toString() {
         final StringBuilder builder = new StringBuilder();
         builder.append(super.toString())
                 .append(" Age: ")
-                .append(getAge());
+                .append(getAge())
+                .append(" Priority: ")
+                .append(getPriority());
 
         return builder.toString();
     }
 
+    /**
+     * Returns true if both persons have the same identity and data fields.
+     * This defines a stronger notion of equality between two persons.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Patient)) {
+            return false;
+        }
+
+        Patient otherPerson = (Patient) other;
+        return otherPerson.getNric().equals(getNric())
+                && otherPerson.getName().equals(getName())
+                && otherPerson.getPhone().equals(getPhone())
+                && otherPerson.getType().equals(getType())
```

Comment:
very minor: maybe can switch line 62 to 59. ie: the type, nric,phone then age then priority

Reply:
agree with simnee on this, would be better to follow attribute order

Reply:
Changed